apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Data;
+using System.Data.Common;
+
+// ReSharper disable UnusedParameter.Local
+
+namespace Google.Cloud.Spanner
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class SpannerParameter : DbParameter
+#if NET451
+        , ICloneable
+#endif
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        public SpannerParameter()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        public SpannerParameter(string spannerColumnName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        /// <param name="sourceDataTableName"></param>
+        public SpannerParameter(string spannerColumnName, string sourceDataTableName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="spannerColumnName"></param>
+        /// <param name="type"></param>
+        public SpannerParameter(string spannerColumnName, SpannerDbType type) : this(spannerColumnName)
```

Comment:
When you do come to implement these fully, I'd try to arrange it for "smaller" constructors to end up calling one "master" constructor which accepts all possible parameters, rather than the other way round. But I understand this isn't a full implementation yet :)